### PR TITLE
Basic sdk features by @gpaquet

### PR DESCRIPTION
**Pull request by daddykotex**

### CODE REVIEW

File: mnubo/services.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+class EventServices(object):
+
+    def __init__(self, api_manager):
+        """ Initializes EventServices with the api manager
+        """
+
+        self.api_manager = api_manager
+
+    def send(self, event):
```
Comment (daddykotex) [inline mnubo/services.py]: We also have a `send` endpoint for specific objects: `/objects/{device_id}/events`. Did you discuss support of this endpoint in this SDK?

File: mnubo/models.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import datetime
+
+
+class Owner(object):
+    """ A mnubo owner object model with the following properties:
+
+    Properties:
+        username: (string)
+        password: (string)
+        registration_date: (datetime)
```
Comment (daddykotex) [inline mnubo/models.py]: should also contain the `event_id`

File: mnubo/api_manager.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+def authenticate(func):
+    def authenticate_and_call(*args):
+        if not args[0].access_token.is_valid():
+            args[0].access_token = args[0].fetch_access_token()
+        return func(*args)
+    return authenticate_and_call
+
+
+class APIManager(object):
```
Comment (daddykotex) [inline mnubo/api_manager.py]: notice there is no `GET` support. This must also have been discussed?

File: mnubo/tests/test_api_manager.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+def test_create_operations():
+
+    response = Response()
+    response._content = '{"access_token":"ACCESS_TOKEN","token_type":"Bearer","expires_in":3887999}'
+    requests.post = MagicMock(return_value=response)
+
+    auth = APIManager('CLIENT_ID', 'CLIENT_SECRET', 'HOSTNAME')
+
+    response = Response()
+    response._content = '{"message": "SUCCESS"}'
```
Comment (daddykotex) [inline mnubo/tests/test_api_manager.py]: I don't see anywhere in the `api_manager` test a validation of the URL that should be called. In this case I'd expect to see something like `requests.expectedUrl=HOSTNAME/api/v3/ROUTE` for example.

Sorry for my python!

File: mnubo/tests/test_models.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+from mnubo.models import Object
+from mnubo.models import AccessToken
+
+
+def test_owner_to_dict():
+    current_time = datetime.datetime.now().isoformat()
+    owner = Owner('USERNAME', 'PASSWORD', current_time, 'EVENT_ID')
+    owner.gender = 'male'
+    owner.age = 30
+    owner.height = 1.80
```
Comment (daddykotex) [inline mnubo/tests/test_models.py]: What happens if I try to do `owner.username="newuser'`. Will it override the value used in the constructor? Just a general question.

**Comment by daddykotex** [comment]
Excellent work btw, just a couple comments/questions!

File: mnubo/tests/test_models.py (hunk 5 above)
Comment (paquetgp) [inline mnubo/tests/test_models.py]: Yes it will.

File: mnubo/api_manager.py (hunk 3 above)
Comment (paquetgp) [inline mnubo/api_manager.py]: When would the GET be needed?

File: mnubo/services.py (hunk 1 above)
Comment (paquetgp) [inline mnubo/services.py]: This endpoint is only used from mobile/device apps according to the documentation.

File: mnubo/models.py (hunk 2 above)
Comment (paquetgp) [inline mnubo/models.py]: Done

File: mnubo/tests/test_api_manager.py (hunk 4 above)
Comment (paquetgp) [inline mnubo/tests/test_api_manager.py]: I'm still working on a way to expect it from the mock.

File: mnubo/api_manager.py (hunk 3 above)
Comment (dmasse76) [inline mnubo/api_manager.py]: There is no GET today as GET can be done through the /search.  However, we will eventually add some basic GET but it is not required at this moment.

File: mnubo/services.py (hunk 1 above)
Comment (dmasse76) [inline mnubo/services.py]: This is right.

File: mnubo/tests/test_api_manager.py (hunk 4 above)
Comment (dmasse76) [inline mnubo/tests/test_api_manager.py]: I'll accept the MR and we can add the URL validation test in a later pull request.

Comment (daddykotex) [inline mnubo/tests/test_api_manager.py]: ok with that!

**Pull request merged**